cqf-fhir-utility/src/main/java/org/opencds/cqf/fhir/utility/visitor/r4/KnowledgeArtifactPackageVisitor.java

```diff
@@ -187,4 +221,368 @@ public static List<BundleEntryComponent> findUnsupportedInclude(
+                                                        .getSystem()
+                                                        .equals(KnowledgeArtifactAdapter.usPhContextTypeUrl)
+                                                && useContext
+                                                        .getCode()
+                                                        .getCode()
+                                                        .equals("reporting")
+                                                && useContext
+                                                        .getValueCodeableConcept()
+                                                        .hasCoding(
+                                                                KnowledgeArtifactAdapter.usPhContextUrl, "triggering"))
+                                        || (useContext
+                                                        .getCode()
+                                                        .getSystem()
+                                                        .equals(KnowledgeArtifactAdapter.usPhContextTypeUrl)
+                                                && useContext
+                                                        .getCode()
+                                                        .getCode()
+                                                        .equals("specification-type")
+                                                && useContext
+                                                        .getValueCodeableConcept()
+                                                        .hasCoding(
+                                                                KnowledgeArtifactAdapter.usPhContextUrl, "program"))))
+                .findFirst();
+        return rootSpecLibrary.orElse(null);
+    }
+
+    private static Parameters getExpansionParams(Library rootSpecificationLibrary, String reference) {
+        Optional<Resource> expansionParamResource = rootSpecificationLibrary.getContained().stream()
+                .filter(contained -> contained.getId().equals(reference))
+                .findFirst();
+        return (Parameters) expansionParamResource.orElse(null);
+    }
+
+    private void checkIfValueSetNeedsCondition(
+            MetadataResource resource, RelatedArtifact relatedArtifact, Repository repository)
+            throws UnprocessableEntityException {
+        if (resource == null
+                && relatedArtifact != null
+                && relatedArtifact.hasResource()
+                && Canonicals.getResourceType(relatedArtifact.getResource()).equals("ValueSet")) {
+            List<MetadataResource> searchResults =
+                    getResourcesFromBundle(searchResourceByUrl(relatedArtifact.getResource(), repository));
+            if (searchResults.size() > 0) {
+                resource = searchResults.get(0);
+            }
+        }
+        if (resource != null && resource.getResourceType() == ResourceType.ValueSet) {
```

Comment:
```suggestion
    private void checkIfValueSetNeedsCondition(
            IBaseResource resource, RelatedArtifact relatedArtifact, Repository repository)
            throws UnprocessableEntityException {
        if (resource == null
                && relatedArtifact != null
                && relatedArtifact.hasResource()
                && Canonicals.getResourceType(relatedArtifact.getResource()).equals(ResourceType.ValueSet.name())) {
            var searchResults = BundleHelper.getEntryResources(SearchHelper.searchRepositoryByCanonicalWithPaging(repository, relatedArtifact.getResourceElement()));
            if (searchResults.size() > 0) {
                resource = searchResults.get(0);
            }
        }
        if (resource != null && resource.fhirType().equals(ResourceType.ValueSet.name())) {
```

Reply:
This removes the need for the getResourcesFromBundle and searchResourceByUrl methods and the ResourceClassMapHelper class.

Given where this method is called though, resource will never be null so this search will never be called anyway. 

Is the intention to handle bundle entries without a resource but perhaps with a fullUrl?

Reply:
This change should also be made in the other two versions of this class.  The way it is currently using the ResourceClassMapHelper will cause Dstu3 and R5 to always fail.

Reply:
Thanks for the feedback @barhodes.

Reply:
Yup.  If we do want to handle bundle entries without a resource that can be handled later.

Reply:
@barhodes removed!